filterExprParser.go

```diff
@@ -0,0 +1,1160 @@
+		return NotExpr{subNot}, err
+	} else if f.Operand != nil {
+		return f.Operand.OutputExpression()
+	} else if f.SubExpr != nil {
+		return f.SubExpr.OutputExpression()
+	} else {
+		return nil, fmt.Errorf("Invalid FECondition %v", f.String())
+	}
+}
+
+type FEOperand struct {
+	BooleanExpr *FEBooleanExpr `@@ |`
+	LHS         *FELhs         `( @@ (`
+	Op          *FECompareOp   `( @@`
+	RHS         *FERhs         `@@ ) | `
+	CheckOp     *FECheckOp     `@@ ) )`
+}
+
+func (feo *FEOperand) String() string {
+	if feo.BooleanExpr != nil {
+		return feo.BooleanExpr.String()
+	} else if feo.CheckOp != nil {
+		return fmt.Sprintf("%v %v", feo.LHS.String(), feo.CheckOp.String())
+	} else if feo.Op != nil {
+		return fmt.Sprintf("%v %v %v", feo.LHS.String(), feo.Op.String(), feo.RHS.String())
+	} else {
+		return "?? (FEOperand)"
+	}
+}
+
+func (f *FEOperand) OutputExpression() (Expression, error) {
+	if f.BooleanExpr != nil {
+		return f.BooleanExpr.OutputExpression()
+	} else if f.LHS != nil {
+		lhsExpr, err := f.LHS.OutputExpression()
+		if err != nil {
+			return nil, err
+		}
+
+		if f.CheckOp != nil {
+			outExpr, err := f.CheckOp.OutputExpression(lhsExpr)
+			return outExpr, err
+		} else if f.Op != nil && f.RHS != nil {
+			rhsExpr, err := f.RHS.OutputExpression()
+			if err != nil {
+				return nil, err
+			}
+			return f.Op.OutputExpression(lhsExpr, rhsExpr)
+		} else {
+			return nil, fmt.Errorf("Invalid FEOperand %v", f.String())
+		}
+	} else {
+		return nil, fmt.Errorf("Invalid FEOperand %v", f.String())
+	}
+}
+
+type FEBooleanExpr struct {
+	BooleanVal  *FEBoolean         `@@ |`
+	BooleanFunc *FEBooleanFuncExpr `@@`
+}
+
+func (be *FEBooleanExpr) String() string {
+	if be.BooleanVal != nil {
+		return be.BooleanVal.String()
+	} else if be.BooleanFunc != nil {
+		return be.BooleanFunc.String()
+	} else {
+		return "?? (FEBooleanExpr)"
+	}
+}
+
+func (f *FEBooleanExpr) OutputExpression() (Expression, error) {
+	if f.BooleanVal != nil {
+		return f.BooleanVal.OutputExpression()
+	} else if f.BooleanFunc != nil {
+		return f.BooleanFunc.OutputExpression()
+	}
+
+	return nil, fmt.Errorf("Invalid FEBooleanExpr %v", f.String())
+}
+
+type FEBoolean struct {
+	TVal  *bool `@"TRUE" |`
+	TVal1 *bool `@"true" |`
+	FVal  *bool `@"FALSE" |`
+	FVal1 *bool `@"false"`
+}
+
+func (feb *FEBoolean) String() string {
+	if feb.TVal != nil && *feb.TVal == true {
+		return "TRUE(bool)"
+	} else if feb.TVal1 != nil && *feb.TVal1 == true {
+		return "true(bool)"
+	} else if feb.FVal != nil && *feb.FVal == true {
+		return "FALSE(bool)"
+	} else if feb.FVal1 != nil && *feb.FVal1 == true {
+		return "false(bool)"
+	}
+	return ""
+}
+
+func (feb *FEBoolean) GetBool() bool {
+	if feb.TVal != nil && *feb.TVal == true {
+		return true
+	} else if feb.TVal1 != nil && *feb.TVal1 == true {
+		return true
+	} else if feb.FVal != nil && *feb.FVal == true {
+		return false
+	} else if feb.FVal1 != nil && *feb.FVal1 == true {
+		return false
+	}
+	return false
```

Comment:
better to return a different, error, state, like String() does?

Reply:
IsSet() is meant for that. I can add a comment.